sstables/types.hh

```diff
@@ -126,16 +126,28 @@ inline std::ostream& operator<<(std::ostream& o, indexable_element e) {
 
 class summary_entry {
 public:
-    dht::token token;
+    int64_t raw_token;
     bytes_view key;
     uint64_t position;
 
+    explicit summary_entry(dht::token token, bytes_view key, uint64_t position)
+            : raw_token(dht::token::to_int64(token))
+            , key(key)
+            , position(position) {
+        assert(!token.is_minimum());
+        assert(!token.is_maximum());
```

Comment:
please replace this by on_internal_error. assert is evil.
By the way to_int64() already checks for is_minimum and is_maximum, and if this is the case, returns numeric_limits<int64_t>::min(). So these assertions/on_internal_error aren't really needed to ensure correctness so maybe they aren't needed at all.

Reply:
See:

```
    // INT64_MIN is not a legal token, but a special value used to represent
    // infinity in token intervals.
    // If a token with value INT64_MIN is generated by the hashing algorithm,
    // the result is coerced into INT64_MAX.
    // (So INT64_MAX is twice as likely as every other token.)
```

but once we normalize it, it's turned into INT64_MAX.

Given that INT64_MAX is a possible token for the hashing, I think it's better to get rid of asserts() and let the dht layer handle it using the dht::normalize() method, which is called automatically on dht::token ctor.